wallet-new/test/unit/TxMetaStorageSpecs.hs

```diff
@@ -0,0 +1,167 @@
+{-# LANGUAGE RankNTypes #-}
+module TxMetaStorageSpecs (txMetaStorageSpecs) where
+
+import           Universum
+
+import           Cardano.Wallet.Kernel.DB.TxMeta
+import           Control.Exception.Safe (bracket, catchJust)
+import qualified Data.List.NonEmpty as NonEmpty
+import           Data.Text.Buildable (build)
+
+import           Formatting (bprint)
+import           Serokell.Util.Text (listJsonIndent)
+import           System.Directory (getTemporaryDirectory, removeFile, withCurrentDirectory)
+import           System.IO.Error (isDoesNotExistError)
+import           Test.Hspec (shouldThrow)
+import           Test.QuickCheck (arbitrary, generate, resize, vectorOf)
+import           Util.Buildable.Hspec
+
+import qualified Cardano.Wallet.Kernel.DB.Sqlite as Storage
+
+-- | Handy combinator which yields a fresh database to work with on each spec.
+withTemporaryDb :: forall m a. (MonadIO m, MonadMask m) => (MetaDBHandle -> m a) -> m a
```

Comment:
Why not use the in memory database? If we're just doing this for testing then touching the disk is going to be way slower than just keeping it in memory. Also simpler.

Reply:
Not sure if I'm interpreting your comment correctly, but what I want to test here is specifically that my Sqlite/beam queries are sound, so disk I/IO is a price I'm willing to pay.

Reply:
Ah, sorry, after caffeine kicked in I see what you mean -- you probably refer to the fact that [sqlite can run in memory](https://www.sqlite.org/inmemorydb.html) and thus might be a nice win to avoid the disk I/O to have faster tests. Makes sense, will look into it 😉